core/modules/startup/focus-handler.js

```diff
@@ -0,0 +1,123 @@
+/*\
+title: $:/plugins/custom/sidebar-focus/startup.js
+type: application/javascript
+module-type: startup
+Handles focus management and spacebar scrolling for sidebar layout
+\*/
+"use strict";
+
+exports.name = "sidebar-focus";
+exports.platforms = ["browser"];
+exports.after = ["story"];
+exports.synchronous = true;
+
+exports.startup = function() {
+	var mainSelector = ".tc-story-river";
+	var secondarySelector = ".tc-secondary-container";
+	
+	// Helper: Check if element is in any secondary container
+	function isInSecondary(element) {
+		var containers = document.querySelectorAll(secondarySelector);
+		return Array.from(containers).some(function(container) {
+			return container === element || container.contains(element);
+		});
+	}
+	
+	// Wait for DOM to be fully ready
+	function initialize() {
+		var main = document.querySelector(mainSelector);
+		
+		if(!main) {
+			setTimeout(initialize, 100);
```

Comment:
IMO if something is activated after `story`, it should not be needed to have a timeout somewhere.

If this timeout is ever activated, there is something fundamentally wrong.